vertical-pod-autoscaler/pkg/utils/vpa/limit_and_request_scaling.go

```diff
@@ -107,7 +115,25 @@ func GetBoundaryRequest(originalRequest, originalLimit, boundaryLimit, defaultLi
 	if originalRequest == nil || originalRequest.Value() == 0 {
 		return boundaryLimit
 	}
-	result, _ := scaleQuantityProportionally(originalRequest /* scaledQuantity */, originalLimit /*scaleBase*/, boundaryLimit /*scaleResult*/, noRounding)
+	result, _ := scaleQuantityProportionallyCPU(originalRequest /* scaledQuantity */, originalLimit /*scaleBase*/, boundaryLimit /*scaleResult*/, noRounding)
+	return result
+}
+
+// GetBoundaryRequestMem returns the boundary (min/max) memory request that can be specified with
+// preserving the original limit to request ratio. Returns nil if no boundary exists
+func GetBoundaryRequestMem(originalRequest, originalLimit, boundaryLimit, defaultLimit *resource.Quantity) *resource.Quantity {
+	if originalLimit == nil || originalLimit.Value() == 0 && defaultLimit != nil {
+		originalLimit = defaultLimit
+	}
+	// originalLimit not set, no boundary
+	if originalLimit == nil || originalLimit.Value() == 0 {
+		return &resource.Quantity{}
+	}
+	// originalLimit set but originalRequest not set - K8s will treat the pod as if they were equal
+	if originalRequest == nil || originalRequest.Value() == 0 {
```

Comment:
Why this is not in `GetBoundaryRequestCPU`?  I think it should be there.

Reply:
It should be there, since `GetBoundaryRequestCPU` is just the original function, but renamed. Nothing was removed from it, rather a copy of it was created and named `GetBoundaryRequestMem` to get the scaled memory values (as opposed to scaled CPU values).